content/blog/api-console-for-data-services-cloud-console.md

```diff
@@ -0,0 +1,200 @@
+---
+title: Using HPE GreenLake Console's API Gateway for Data Services Cloud Console
+date: 2021-07-29T14:28:17.779Z
+featuredBlog: false
+priority: 1
+author: Ron Dharma
+authorimage: https://gravatar.com/avatar/8102f2adeef249065ccb9c43b8866d17?s=192
+thumbnailimage: /img/dscc-icon.png
+tags:
+  - data-services-cloud-console
+---
+## Secured, Yet Agile
+
+A major guiding principle in the creation of the Application Programming Interface (API) for Data Services Cloud Console (DSCC) from Hewlett Packard Enterprise (HPE) is security. However, to be able to be used by applications or tools that rely on the API to extend their features using DSCC, the API must also be flexible. To provide both security and flexibility, the DSCC REST API uses the 0Auth 2.0 authentication flow based on the client credential, which generates a limited lifetime access token. This access token is then embedded in the header of each REST API request as the authorization bearer.
+
+This blog will walk through the essential steps required to exercise or experiment with the DSCC REST API.
+
+### Authentication Process to Obtain the Access Token
+
+The DSCC public API relies on an OAuth 2.0 third party authorization framework on behalf of the resource owner (HPE GreenLake Console's user) for security. The user starts by logging and authenticating into HPE GreenLake Console, which is authenticated by the Identity Provider (validated through username, password, or Multi Factor Authentication). Using the API gateway menu in GreenLake Console, a customer registers their client application to obtain the OAuth 2.0 API client credentials (client-id and client-secret). This association allows the user to obtain the access token from the menu, and user can then use the access token inside the token bearer field (header) with any REST API request. This action allows any client application or script to perform any API request to the correct instance of DSCC.
+
+![client-credential application flow](/img/greenlake-api-access-flow.png "obtain client-id and client-secret")
+
+The access token have a limited lifetime (about 720 seconds or 2 hours). Once it expires, the client application must use the obtained client-id and client-secret to generate a new access token. One indication of the expiration of access-token, the request to DSCC API will return response error: '401 Unauthorized HTTP.'  If the client application generates new access token prior to it expires, then the current access-token will be invalidated or treated as not authorized. 
```

Comment:
Change "The access token have" to "The access token has"
Remove hyphen between "access" and "token" on the second and third lines
Add "the" prior to "access token, the request"
Add "a" prior to "new access token prior to"
Add "when" prior to "it expires"
So, the last sentence should read "If the client application generates a new access token prior to when it expires, then the current access token will be invalidated or treated as not authorized."